packages/backend/src/ee/services/ai/agents/agent.ts

```diff
@@ -164,6 +164,18 @@ export const generateAgentResponse = async ({
                         }),
                     );
                 }
+                if (step.toolResults && step.toolResults.length > 0) {
+                    await Promise.all(
+                        step.toolResults.map(async (toolResult) => {
+                            await dependencies.storeToolResult({
+                                promptUuid: args.promptUuid,
+                                toolCallId: toolResult.toolCallId,
+                                toolName: toolResult.toolName,
+                                result: toolResult.result,
+                            });
+                        }),
+                    );
+                }
```

Comment:
`storeToolResults` would be nice to batch store, in a single TX

Reply:
ah yes ! ty